packages/cloud_firestore/ios/Classes/CloudFirestorePlugin.m

```diff
@@ -145,7 +146,9 @@ - (void)writeValue:(id)value {
     [self writeUTF8:documentPath];
   } else if ([value isKindOfClass:[NSData class]]) {
     NSData *blob = value;
-    [super writeValue:blob.bytes];
+    [self writeByte:BLOB];
+    [self writeSize:blob.length];
+    [self writeData:blob.bytes];
```

Comment:
Just `blob` here. `writeData` takes an `NSData*`. I guess this works by accident, because `blob.bytes` gives you a `void*` to the same memory location.